pystac/item_collection.py

```diff
@@ -0,0 +1,125 @@
+from copy import deepcopy
+from pystac.errors import STACTypeError
+from typing import Any, Dict, Iterator, List, Optional, Sized, Iterable
+
+import pystac
+from pystac.utils import make_absolute_href, is_absolute_href
+from pystac.serialization.identify import identify_stac_object_type
+
+
+class ItemCollection(Sized, Iterable[pystac.Item]):
```

Comment:
In the latest iteration in pystac-client I had links as a member of ItemCollection:
https://github.com/stac-utils/pystac-client/blob/mah/updates/pystac_client/item_collection.py#L24

Here it would just be included in extra_fields, but only just as JSON, whereas with a Links array of Link types we can use the PySTAC Link object which is sorta nice.  But that implies that an ItemCollection is it's own defined Object, but we don't have it defined anywhere as having Links outside the API search response.

pystac-client doesn't need them to be Link types because it uses the raw JSON pagination links when paging so while this isn't required it does seem useful.

The most common way of using this is through pystac-client or just for opening up saved search results, and being able to understand the included links would be a nice feature.

Reply:
This makes sense to me, let's keep the basic functionality. We can extend it as needed.